### PR TITLE
initial implementation of `inspect`

### PR DESCRIPTION
**Pull request by shadeofblue**

### CODE REVIEW

**Review by approxit — COMMENTED** [review]
Again, too complex stuff happening in property lookups.

File: dapp_manager/inspect/__main__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    @staticmethod
+    def _get_template(name: str) -> Template:
+        lookup = TemplateLookup(directories=[Path(__file__).parent / "templates"])
+        return lookup.get_template(name)
+
+    @property
+    def gaom(self):
+        if not self._gaom:
+            data = requests.get(f"{self.api_address}/gaom/")
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply dapp_manager/inspect/__main__.py:24]: well, "nobody" is a strong way of putting it... I respect your preference but please refrain from making such general statements, especially if you know that to be _obviously_ false...

File: pyproject.toml [hunk 2]
```diff
@@ -35,7 +35,8 @@ psutil = "^5.9"
 appdirs = "^1.4"
 click = "^7.0"  # requires bump to goth's dependencies https://github.com/golemfactory/goth/issues/605
 pydantic = "^1.9"
-dapp-runner = "^0.1.1"
+dapp-runner = { git = "https://github.com/golemfactory/dapp-runner.git", branch = "main" }
+mako = "^1.2.4"
```
[the remark this answers is not in the document]
Comment (shadeofblue) [reply pyproject.toml:39]: just that I get the impression that Mako is a more lightweight, and thus, a better alternative for a simple use that we have here... but no other strong argument

File: dapp_manager/inspect/__main__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shadeofblue) [reply dapp_manager/inspect/__main__.py:24]: but honestly, the whole point of a python property is to present an attribute api to something that requires programmatic resolution... I don't see a contradiction here - espeically if the result is cached after the first try...

to do it another way would be awkward ... like what would you call the function? `get_or_retrieve_a_cached_gaom` ?

**Review by approxit — APPROVED** [review]

**Pull request merged**